packages/time-picker/src/TimePicker/TimePicker.tsx

```diff
@@ -0,0 +1,202 @@
+
+import { Item } from "@react-stately/collections"
+import { useMenuTriggerState } from "@react-stately/menu"
+import {
+  useTimeFieldState,
+  TimeFieldStateOptions,
+} from "@react-stately/datepicker"
+
+import chevronDown from "@kaizen/component-library/icons/chevron-down.icon.svg"
+import chevronUp from "@kaizen/component-library/icons/chevron-up.icon.svg"
+import classNames from "classnames"
+import { formatDateToTime, getAllTimeOptions, TIME_OPTION } from "./utils"
+import { DateSegment, Menu, Button, Popover } from "./components"
+import styles from "./TimePicker.module.scss"
+import { TimeValue } from "./types"
+import { convertTimeToZonedDateTime } from "./utils/convertTimeToZonedDateTime"
+
+export type StatusType = "default" | "error"
+
+const isZonedDateTimeObject = (
+  timeValue: TimeValue
+): timeValue is ZonedDateTime => "toDate" in timeValue
+export interface TimePickerProps
+  extends Omit<
+    TimeFieldStateOptions,
+    "errorMessage" | "validationState" | "value" | "onChange" | "label"
+  > {
+  id: string
+  label: string
+  onChange: (date: Date) => void
+  value: Date | undefined
+  /**
+   * Supply timeZone in IANA format, i.e. "Australia/Melbourne"
+   */
+  timeZone?: string
+  dropdownIncrements?: number
+  status?: StatusType
+  validationMessage?: React.ReactNode
+}
+
+export const TimePicker: React.VFC<TimePickerProps> = ({
+  status = "default",
+  validationMessage,
+  id,
+  isDisabled,
+  value,
+  onChange,
+  label,
+  timeZone = getLocalTimeZone(),
+  locale,
+  dropdownIncrements,
+
+  ...restProps
+}: TimePickerProps) => {
+  const handleOnChange = (timeValue: TimeValue): void => {
+    // onChange does not fire until user interacts with all placeholders
+    // if user interacts with spin buttons, timeValue is Time type, which cannot be converted directly into Date object
+    // if user interacts with MenuItems, it has to be CalendarDateTime or ZonedDateTime, as Time objects cannot be formatted properly to display in a list
+    // console.log("STATE", state.dateValue, timeValue)
+    if (isZonedDateTimeObject(timeValue)) {
+      onChange(timeValue.toDate())
+    } else {
+      const today = now(timeZone).toDate()
+      const date = convertTimeToZonedDateTime({
+        date: value ?? today,
+        hour: timeValue.hour,
+        minutes: timeValue.minute,
+        timeZone,
+      }).toDate()
+      onChange(date)
+    }
+  }
+
+  const state = useTimeFieldState({
+    ...restProps,
+    // @ts-expect-error controlled values should not be undefined and library throws warning if such is supplied for value,
+    // however library does not allow for null values despite handling properly
+    // https://github.com/adobe/react-spectrum/blob/main/packages/%40react-stately/utils/src/useControlledState.ts#L23
+    // https://github.com/adobe/react-spectrum/blob/main/packages/%40react-stately/datepicker/src/useTimeFieldState.ts
+    value: value ? parseAbsolute(value.toISOString(), timeZone) : null,
```

Comment:
Why about falling back to an empty string instead of `null`?

Reply:
We wanted the behaviour of having an empty time picker with no value set if there was no default value